airflow/utils/scheduler.py

```diff
@@ -0,0 +1,569 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from abc import ABCMeta, abstractmethod
+from datetime import datetime
+import logging
+import os
+import re
+
+from airflow.exceptions import AirflowException
+from airflow.utils.logging import LoggingMixin
+
+
+class SimpleDag(object):
```

Comment:
I'm having a try at this, but it looks like it makes the `DAG` class a bit messier.